src/Lecture2.hs

```diff
@@ -74,7 +79,14 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt ix xs
+  | null xs = (Nothing, [])
+  | ix < 0 || ix >= length xs = (Nothing, xs)
+  | ix == 0 = (Just (head xs), tail xs)
+  | ix == length xs - 1 = (Just (last xs), init xs)
+  | otherwise = (Just (xs !! ix), take ix xs ++ drop (ix + 1) xs)
```

Comment:
That's a correct implementation 👍🏻 
Alternatively, you can use a standard function [splitAt](https://hackage.haskell.org/package/base-4.16.0.0/docs/Prelude.html#v:splitAt) instead of using `take` + `drop` 🔃